File: app/src/main/java/net/opendasharchive/openarchive/db/Collection.java
package net.opendasharchive.openarchive.db;

import com.orm.SugarRecord;

import java.util.Date;
import java.util.List;

public class Collection extends SugarRecord {


    public long projectId;

    public Date uploadDate;

    public String serverUrl;

    public long getProjectId() {
        return projectId;
    }

    public void setProjectId(long projectId) {
        this.projectId = projectId;
    }

    public Date getUploadDate() {
        return uploadDate;
    }

    public void setUploadDate(Date uploadDate) {
        this.uploadDate = uploadDate;
    }

    public String getServerUrl() {
        return serverUrl;
    }

    public void setServerUrl(String serverUrl) {
        this.serverUrl = serverUrl;
    }

    public static List<Collection> getAllAsList() {
        return Collection.find(Collection.class,null,null,null,"ID DESC",null);
    }

    public static List<Collection> getAllAsListByProject(long projectId) {
        String[] values = {projectId+""};
        return Collection.find(Collection.class,"PROJECT_ID = ?",values,null,"ID DESC",null);
    }
}

File: app/src/main/java/net/opendasharchive/openarchive/db/Collection.kt
package net.opendasharchive.openarchive.db

import com.orm.SugarRecord
import java.util.*

data class Collection(
    var projectId: Long? = null,
    var uploadDate: Date? = null,
    var serverUrl: String? = null
) : SugarRecord() {

    companion object {

        fun getAll(): List<Collection> {
            return find(Collection::class.java, null, arrayOf(),
                null, "id DESC", null)
        }
    }

    val media: List<Media>
        get() = find(Media::class.java, "collection_id = ?", arrayOf(id.toString()), null, "status, id DESC", null)


    override fun delete(): Boolean {
        media.forEach {
            it.delete()
        }

        return super.delete()
    }
}